homeassistant/components/amcrest/binary_sensor.py

```diff
@@ -105,35 +105,21 @@ class AmcrestBinarySensor(BinarySensorEntity):
 
     def __init__(self, name, device, sensor_type):
         """Initialize entity."""
-        self._name = f"{name} {BINARY_SENSORS[sensor_type][SENSOR_NAME]}"
+        self._attr_name = f"{name} {BINARY_SENSORS[sensor_type][SENSOR_NAME]}"
         self._signal_name = name
         self._api = device.api
         self._sensor_type = sensor_type
         self._state = None
-        self._device_class = BINARY_SENSORS[sensor_type][SENSOR_DEVICE_CLASS]
+        self._attr_device_class = BINARY_SENSORS[sensor_type][SENSOR_DEVICE_CLASS]
         self._event_code = BINARY_SENSORS[sensor_type][SENSOR_EVENT_CODE]
         self._unsub_dispatcher = []
-
-    @property
-    def should_poll(self):
-        """Return True if entity has to be polled for state."""
-        return self._sensor_type in BINARY_POLLED_SENSORS
-
-    @property
-    def name(self):
-        """Return entity name."""
-        return self._name
+        self._attr_should_poll = self._sensor_type in BINARY_POLLED_SENSORS
 
     @property
     def is_on(self):
         """Return if entity is on."""
         return self._state
```

Comment:
You can set `self._attr_is_on` instead of setting `self._state` where that is used, and this property can be removed as well

Reply:
Done

Reply:
No?

Reply:
Just realised you are in another file, I have made these changes, but I basically only wanted to change sensor.py for this PR, and then copy the final result (all the good comments from Frenck) to the other files.